src/main/java/htsjdk/variant/vcf/VCFFileReader.java

```diff
@@ -268,25 +270,40 @@ public static IntervalList fromVcf(final VCFFileReader vcf, final boolean includ
      * @return an IntervalList constructed from input vcf
      */
     public static IntervalList toIntervalList(final VCFFileReader vcf, final boolean includeFiltered) {
+        final Tuple<SAMFileHeader, Iterator<Interval>> samFileHeaderIteratorTuple = toIntervals(vcf, includeFiltered);
+        final IntervalList intervalList = new IntervalList(samFileHeaderIteratorTuple.a);
+        samFileHeaderIteratorTuple.b.forEachRemaining(intervalList::add);
+        return intervalList;
+    }
+
+    /**
+     * Converts a {@link VCFFileReader} to a Tuple consisting of a SamFileHeader and an iterator of Intervals.
+     * The name field of the IntervalList is taken from the ID field
+     * of the variant, if it exists. If not, creates a name of the format interval-n where n is a running number that increments
+     * only on un-named intervals. Will use a "END" tag in the INFO field as the end of the interval (if exists).
+     *
+     * @param vcf the vcfReader to be used for the conversion
+     * @return a Tuple<SamFileHeader,Iterator<Interval>> constructed from input vcf
+     */
+    public static Tuple<SAMFileHeader,Iterator<Interval>> toIntervals(final VCFFileReader vcf, final boolean includeFiltered) {
 
         //grab the dictionary from the VCF and use it in the IntervalList
         final SAMSequenceDictionary dict = vcf.getFileHeader().getSequenceDictionary();
         final SAMFileHeader samFileHeader = new SAMFileHeader();
         samFileHeader.setSequenceDictionary(dict);
-        final IntervalList list = new IntervalList(samFileHeader);
-
-        int intervals = 0;
-        for (final VariantContext vc : vcf) {
-            if (includeFiltered || !vc.isFiltered()) {
-                String name = vc.getID();
-                final Integer intervalEnd = vc.getCommonInfo().getAttributeAsInt(VCFConstants.END_KEY, vc.getEnd());
-                if (VCFConstants.EMPTY_ID_FIELD.equals(name) || name == null)
-                    name = "interval-" + (++intervals);
-                list.add(new Interval(vc.getContig(), vc.getStart(), intervalEnd, false, name));
+
+        final AtomicReference<Integer> intervals = new AtomicReference<>(1);
```

Comment:
Using AtomicReference<Integer> is weird and kind of gross but lets us write an otherwise nice stream expression instead of a nasty iterator subclass so I understand the reasoning.  Why not use AtomicInteger instead though?  Then you can use getAndIncrement down below as well.

Reply:
Rename to intervalCount or numIntervals.

Reply:
yup.